tools/graphiteindex/src/test/java/com/criteo/biggraphite/graphiteindex/MetricsIndexQueriesTest.java

```diff
@@ -0,0 +1,126 @@
+package com.criteo.biggraphite.graphiteindex;
+
+import static org.junit.Assert.assertEquals;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.tuple.Pair;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+
+/**
+ * Tests queries against an in-memory metrics index.
+ * Offset values are generated using the path's hashCode.
+ */
+@RunWith(Parameterized.class)
+public class MetricsIndexQueriesTest {
+    private static Set<String> setOf(String... xs) {
+        return new HashSet<>(Arrays.asList(xs));
+    }
+
+    public static final Object[][] testQueries = {
+        {
+            "a.single.metric",
+            setOf(
+                "a.single.metric"
+            ),
+        },
+        {
+            "a.few.metrics.*",
+            setOf(
+                "a.few.metrics.a",
+                "a.few.metrics.b",
+                "a.few.metrics.c"
+            ),
+        },
+        {
+            "lots.of.wildcards.*****",
+            setOf(
+                "lots.of.wildcards.a",
+                "lots.of.wildcards.b",
+                "lots.of.wildcards.c"
+            ),
+        },
+        {
+            "metrics.match{ed_by,ing}.s[ao]me.regexp.?[0-9]",
```

Comment:
how dos that work ? does lucene already handle this ?

Reply:
it uses regex-based queries, and yes of course :-)